test/e2e/framework/util.go

```diff
@@ -421,7 +421,7 @@ func hasReplicationControllersForPod(rcs *api.ReplicationControllerList, pod api
 // that it requires the list of pods on every iteration. This is useful, for
 // example, in cluster startup, because the number of pods increases while
 // waiting.
-func WaitForPodsRunningReady(ns string, minPods int32, timeout time.Duration) error {
+func WaitForPodsRunningReady(ns string, minPods int32, timeout time.Duration, countSuccessAsReady bool) error {
```

Comment:
I understand why you introduced this new boolean here, but could you please add a comment for the engineers in the future. 

Also I am thinking splitting this single method to two separate ones, one for WaitForPodsRunningReady, another one for WaitForPodsSucceeded, the test owner / writer should understand their test workloads to call the right functions. WDYT?